rotkehlchen/globaldb/updates.py

```diff
@@ -300,74 +305,219 @@ def _parse_full_insert(self, insert_text: str) -> AssetData:
+        """Process the insertion of a new asset_collection"""
+        collection_match = self.assets_collection_re.match(action)
+        if collection_match is None:
+            raise DeserializationError(
+                f'At asset DB update could not parse asset collection data out of {action}',
+            )
+
+        groups = collection_match.groups()
+        if len(groups) != 3:
+            raise DeserializationError(
+                f'At asset DB update could not parse asset collection data out of {action}',
+            )
+
+        try:
+            with connection.savepoint_ctx() as cursor:
+                cursor.execute(action)
+        except sqlite3.Error:
+            try:
+                with connection.savepoint_ctx() as cursor:
+                    cursor.execute(full_insert)
+            except sqlite3.Error as e:
+                self.msg_aggregator.add_warning(
+                    f'Failed to edit or add asset collection with name {groups[1]} and id '
+                    f'{groups[0]}. Error: {str(e)}',
+                )
+
+    def _process_multiasset_mapping(
+            self,
+            connection: 'DBConnection',
+            action: str,
+            full_insert: str,
+    ) -> None:
+        """
+        Process the insertion of a new asset_collection mapping
+        May raise:
+        - DeserializationError
+        - UnknownAsset
+        """
+        collection_match = self.multiasset_mappings_re.match(action)
+        if collection_match is None:
+            raise DeserializationError(
+                f'At asset DB update could not parse asset collection data out of {action}',
+            )
+
+        groups = collection_match.groups()
+        if len(groups) != 2:
+            raise DeserializationError(
+                f'At asset DB update could not parse asset collection data out of {action}',
+            )
+
+        # check that the asset exists and so does the collection
+        Asset(groups[1]).check_existence()
+        with connection.cursor() as cursor:
+            cursor.execute(
+                'SELECT COUNT(*) FROM assets_collections WHERE id=?',
+                (self._parse_value(groups[0]),),
+            )
+            if cursor.fetchone()[0] != (1,):
+                raise DeserializationError(
+                    f'Tried to add asset to collection with id {groups[0]} but it does not exist',
+                )
+
+        try:
+            with connection.savepoint_ctx() as cursor:
+                cursor.execute(action)
+        except sqlite3.Error:
+            try:
+                with connection.savepoint_ctx() as cursor:
+                    cursor.execute(full_insert)
+            except sqlite3.Error as e:
+                self.msg_aggregator.add_warning(
+                    f'Failed to edit asset collection mapping with asset {groups[1]} '
+                    f'and id {groups[0]}. Error: {str(e)}',
+                )
+
+    def _handle_asset_update(
```

Comment:
Kinda nitpicky, but why is this method called `_handle...` while the other two `_process...` ?

Reply:
It was intentionally because this one is bit more complex and not just reading the info and putting it in the database

Reply:
Alright then :+1: